Deprecate ec2_vpc module

willthames opened the pull request:
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_vpc

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel c5d3aac822) last updated 2017/01/16 11:38:57 (GMT +1000)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY
The deprecation of ec2_vpc module has been discussed for
2 years and is causing duplication of effort as changes
are implemented for ec2_vpc rather than for the newer
alternatives

As mentioned in ansible/ansible-modules-core#995

willthames commented:
@bcoca @defionscode you suggested ec2_vpc was to be deprecated.

zyv commented:
Argh, I wish I saw this earlier :-( implemented a lot on the basis of `ec2_vpc`, because I didn't even realise that `ec2_vpc_net` exists...

willthames commented:
@zyv you and me both... (and doubtless many more)

gundalow commented:
@willthames 

1) Can you please update `CHANGELOG.md` to add note in to say this has been deprecated, see 2.2 
release notes for an eample
2) Please also add the following 
`deprecated: Deprecated in 2.3. Use M(whatmoduletouseinstead) instead.`

gundalow reviewed (changes requested):
needs_revision

bcoca commented:
FYI, to deprecate a module you need to:

- rename with leading _ 
- update description with deprecated field and point at modules taking over the function
- new in 2.3: update metadata

if you have any doubts look at the old openstack modules

Commit:
Deprecate ec2_vpc module

The deprecation of ec2_vpc module has been discussed for
2 years and is causing duplication of effort as changes
are implemented for ec2_vpc rather than for the newer
alternatives

Commit:
Improve module deprecation documentation

Update the developing modules documentation with the latest
instructions on how to deprecate a module.

willthames commented:
I've updated the module to meet the undocumented deprecation instructions.

I've added a second commit that documents the deprecation instructions. I can separate that into a new PR if needed.

gundalow commented:
Thanks :)

mattclay commented on lib/ansible/modules/cloud/amazon/_ec2_vpc.py:
```diff
@@ -25,6 +25,10 @@
     - Create or terminates AWS virtual private clouds.  This module has a dependency on python-boto.
 version_added: "1.4"
+deprecated: >-
+  Deprecated in 2.3. Use M(epc_vpc_net) along with supporting modules including
```
@willthames It looks like a typo made it through here. Shouldn't that be `ec2_vpc_net` not `epc_vpc_net`?
